test/helpers/TestFileLoader.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+export namespace TestFileLoader {
+  export type FilterOption = { exclude?: string[]; include?: string[] };
+
+  const filter = (files: string[], options: FilterOption): string[] => {
+    return files.filter((file) => {
+      if (options.include) return isIncluded();
+      if (options.exclude) return !isExcluded();
+      return true;
+
+      function isExcluded() {
+        return options.exclude?.some((exPath) => file.includes(exPath));
+      }
+
+      function isIncluded() {
+        return options.include?.some((incPath) => file.includes(incPath));
+      }
```

Comment:
`file` includes `__dirname`. This is likely to cause unexpected problems, so please match the name with the rest of the file except `__dirname`.

```ts
// example
file.split(__dirname).at(0)?.includes(incPath)
```